build/templates/session.py.mako

```diff
@@ -169,6 +183,11 @@ constructor_params = helper.filter_parameters(init_function, helper.ParameterUsa
             raise AttributeError("'{0}' object has no attribute '{1}'".format(type(self).__name__, key))
         object.__setattr__(self, key, value)
 
+% if config['use_session_lock']:
+    def lock(self):  # TODO(texasaggie97) Need to figure out how to document this
```

Comment:
how come there's no public `unlock()`?

Reply:
`lock()` is the context manager. `lock_session()` and `unlock_session()` are the functions.

Reply:
With `initiate()` we return the context manager and we have an `abort()` and that's it.
This is different, we have `lock()`which returns the context manager, plus `lock_session()` and `unlock_session()`. I'm guessing this has to do with returning a bool for `caller_has_lock`.

I'd like to see if this can be simplified in a way that makes sense.

Reply:
* Made `lock_session()` private
* Change `unlock_session()` to `unlock()`
* `_Lock()` context manager only unlocks
* `lock()` will call `_lock_session()` and then return `_Lock()` context manager

These changes are to support both
``` python
with session.lock():
   do_something(session)
```
and
``` python
session.lock()
do_something(session)
session.unlock()
```